src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -0,0 +1,1046 @@
+        layer_past=None,
+        head_mask=None,
+        use_cache=False,
+        output_attentions=False,
+    ):
+
+        query = self.q_proj(hidden_states)
+        key = self.k_proj(hidden_states)
+        value = self.v_proj(hidden_states)
+
+        query = self._split_heads(query, self.num_attention_heads, self.head_dim, self.rotary)
+        key = self._split_heads(key, self.num_attention_heads, self.head_dim, self.rotary)
+        value = self._split_heads(value, self.num_attention_heads, self.head_dim, False)
+
+        if self.rotary:
+            seq_len = key.shape[1]
+            offset = 0
+            if layer_past is not None:
+                offset = layer_past[0].shape[-2]
+                seq_len += offset
+            if self.rotary_dim is not None:
+                k_rot = key[:, :, :, :self.rotary_dim]
+                k_pass = key[:, :, :, self.rotary_dim:]
+
+                q_rot = query[:, :, :, :self.rotary_dim]
+                q_pass = query[:, :, :, self.rotary_dim:]
+
+                sincos = fixed_pos_embedding(k_rot, 1, seq_len=seq_len)
+                k_rot = apply_rotary_pos_emb(k_rot, sincos, offset=offset)
+                q_rot = apply_rotary_pos_emb(q_rot, sincos, offset=offset)
+
+                key = torch.cat([k_rot, k_pass], dim=-1)
+                query = torch.cat([q_rot, q_pass], dim=-1)
+            else:
+                sincos = fixed_pos_embedding(key, 1, seq_len=seq_len)
+                key = apply_rotary_pos_emb(key, sincos, offset=offset)
+                query = apply_rotary_pos_emb(query, sincos, offset=offset)
+            key = key.permute(0, 2, 1, 3)
+            query = query.permute(0, 2, 1, 3)
+
+        if layer_past is not None:
+            past_key = layer_past[0]
+            past_value = layer_past[1]
+            key = torch.cat((past_key, key), dim=-2)
+            value = torch.cat((past_value, value), dim=-2)
+
+        if use_cache is True:
+            present = (key, value)
+        else:
+            present = None
+
+        query_length, key_length = query.size(-2), key.size(-2)
+        causal_mask = self.bias[:, :, key_length - query_length : key_length, :key_length]
+
+        attn_output, attn_weights = self._attn(
+            query, key, value, causal_mask, self.masked_bias, self.attn_pdrop, attention_mask, head_mask, self.scale_attn
+        )
+
+        attn_output = self._merge_heads(attn_output, self.num_attention_heads, self.head_dim)
+        attn_output = self.out_proj(attn_output)
+        attn_output = self.resid_pdrop(attn_output)
+
+        outputs = (attn_output, present)
+        if output_attentions:
+            outputs += (attn_weights,)
+
+        return outputs  # a, present, (attentions)
+
+class GPTJAttention(nn.Module):
+    def __init__(self, config, layer_id=0):
+        super().__init__()
+        self.layer_id = layer_id
+        self.attention = GPTJSelfAttention(config)
+        
+    def forward(
+        self,
```

Comment:
Similar comment as above, the `GPTJAttention` could be removed and just use `GPTJSelfAttention` in `GPTJBlock`.